crates/polars-ops/src/frame/pivot/positioning.rs

```diff
@@ -210,6 +210,26 @@ pub(super) fn compute_col_idx(
             let ca = column_agg_physical.bit_repr_large();
             compute_col_idx_numeric(&ca)
         },
+        Struct(_) => {
+            let mut col_to_idx = PlHashMap::with_capacity(HASHMAP_INIT_SIZE);
+            let mut idx = 0 as IdxSize;
+            let ca = column_agg_physical.struct_()?;
+            (0..(column_agg_physical.len()))
```

Comment:
using `phys_iter`, as is done in the arm below, wouldn't work here, because then iteration would be over, for example

- Struct(2, StructArray[{bar: [121], bart: [121]}, {bar: [121], bart: [121]}, {bar: [120], bart: [120]}, {bar: [120], bart: [120]}], [Field { name: "bar", dtype: String }, Field { name: "bart", dtype: String }]): 2
- Struct(3, StructArray[{bar: [121], bart: [121]}, {bar: [121], bart: [121]}, {bar: [120], bart: [120]}, {bar: [120], bart: [120]}], [Field { name: "bar", dtype: String }, Field { name: "bart", dtype: String }]): 3,
- Struct(0, StructArray[{bar: [121], bart: [121]}, {bar: [121], bart: [121]}, {bar: [120], bart: [120]}, {bar: [120], bart: [120]}], [Field { name: "bar", dtype: String }, Field { name: "bart", dtype: String }]): 0,
- Struct(1, StructArray[{bar: [121], bart: [121]}, {bar: [121], bart: [121]}, {bar: [120], bart: [120]}, {bar: [120], bart: [120]}], [Field { name: "bar", dtype: String }, Field { name: "bart", dtype: String }]): 1
}

Reply:
I optimized this and extended that optimization to other dtypes as well. 

We must do the same for `row_index`, want to take that one. It is similar.

Reply:
thanks for adding a commit here! yes, definitely, have added to my backlog, will do it separately if it's ok (as far as I can tell it it's orthogonal to the linked issue) and opened a separate issue about it https://github.com/pola-rs/polars/issues/14101